File: zz-tests_bats/test-fixtures/arboretum/sample.php
<?php
// Sample PHP file.

namespace SampleNs;

const DEFAULT_NAME = "world";

function shout(string $s): string {
    return strtoupper($s);
}

interface Greeter {
    public function greet(string $name): string;
}

trait Countable {
    public int $count = 0;

    public function tick(): void {
        $this->count++;
    }
}

class HelloGreeter implements Greeter {
    use Countable;

    public string $salutation = "hi";

    public function greet(string $name): string {
        $this->tick();
        return "{$this->salutation}, {$name}";
    }
}

File: zz-tests_bats/test-fixtures/arboretum/sample.go
package outline

import (
	"fmt"
	"strings"
)

// Greeter wraps a salutation.
type Greeter struct {
	Salutation string
	count      int
}

// Greet returns the greeting for name.
func (g *Greeter) Greet(name string) string {
	g.count++
	return fmt.Sprintf("%s, %s!", g.Salutation, strings.ToUpper(name))
}

func (g *Greeter) Count() int { return g.count }

func NewGreeter(s string) *Greeter {
	return &Greeter{Salutation: s}
}

const DefaultSalutation = "Hello"

var globalGreeter = NewGreeter(DefaultSalutation)

type Renamer interface {
	Rename(old, new string) error
}
